pkgs/applications/graphics/gimp/default.nix

```diff
@@ -135,84 +173,144 @@ stdenv.mkDerivation (finalAttrs: {
+    ++ lib.optionals stdenv.hostPlatform.isDarwin [
+      "-Dalsa=disabled"
+      "-Djavascript=disabled"
     ];
 
-  enableParallelBuilding = true;
-
+  # on Linux, unable to find icons
   doCheck = true;
 
   env = {
-    NIX_CFLAGS_COMPILE = toString (
-      [ ]
-      ++ lib.optionals stdenv.cc.isGNU [ "-Wno-error=incompatible-pointer-types" ]
-      ++ lib.optionals stdenv.hostPlatform.isDarwin [ "-DGDK_OSX_BIG_SUR=16" ]
-    );
+    # The check runs before glib-networking is registered
+    GIO_EXTRA_MODULES = "${glib-networking}/lib/gio/modules";
+
+    NIX_CFLAGS_COMPILE = lib.optionalString stdenv.hostPlatform.isDarwin "-DGDK_OSX_BIG_SUR=16";
 
     # Check if librsvg was built with --disable-pixbuf-loader.
     PKG_CONFIG_GDK_PIXBUF_2_0_GDK_PIXBUF_MODULEDIR = "${librsvg}/${gdk-pixbuf.moduleDir}";
   };
 
-  preConfigure = ''
-    # The check runs before glib-networking is registered
-    export GIO_EXTRA_MODULES="${glib-networking}/lib/gio/modules:$GIO_EXTRA_MODULES"
+  postPatch = ''
+    patchShebangs \
+      app/tests/create_test_env.sh \
+      tools/gimp-mkenums
+
+    # GIMP is executed at build time so we need to fix this.
+    # TODO: Look into if we can fix the interp thing.
+    chmod +x plug-ins/python/{colorxhtml,file-openraster,foggify,gradients-save-as-css,histogram-export,palette-offset,palette-sort,palette-to-gradient,python-eval,spyro-plus}.py
+    patchShebangs \
+      plug-ins/python/{colorxhtml,file-openraster,foggify,gradients-save-as-css,histogram-export,palette-offset,palette-sort,palette-to-gradient,python-eval,spyro-plus}.py
+  '';
+
+  preBuild = ''
+    # Our gobject-introspection patches make the shared library paths absolute
+    # in the GIR files. When running GIMP in build or check phase, it will try
+    # to use plug-ins, which import GIMP introspection files which will try
+    # to load the GIMP libraries which will not be installed yet.
+    # So we need to replace the absolute path with a local one.
+    # We are using a symlink that will be overridden during installation.
+    mkdir -p "$out/lib"
+    ln -s "$PWD/libgimp/libgimp-3.0.so.0" "$out/lib/libgimp-3.0.so.0"
+    ln -s "$PWD/libgimpbase/libgimpbase-3.0.so.0" "$out/lib/libgimpbase-3.0.so.0"
+    ln -s "$PWD/libgimpcolor/libgimpcolor-3.0.so.0" "$out/lib/libgimpcolor-3.0.so.0"
+    ln -s "$PWD/libgimpconfig/libgimpconfig-3.0.so.0" "$out/lib/libgimpconfig-3.0.so.0"
+    ln -s "$PWD/libgimpmath/libgimpmath-3.0.so.0" "$out/lib/libgimpmath-3.0.so.0"
+    ln -s "$PWD/libgimpmodule/libgimpmodule-3.0.so.0" "$out/lib/libgimpmodule-3.0.so.0"
+  '';
+
+  preCheck = ''
+    # Avoid “Error retrieving accessibility bus address”
+    export NO_AT_BRIDGE=1
+    # Fix storing recent file list in tests
+    export HOME="$TMPDIR"
+    export XDG_DATA_DIRS="${glib.getSchemaDataDirPath gtk3}:${adwaita-icon-theme}/share:$XDG_DATA_DIRS"
+  '';
+
+  checkPhase = ''
+    runHook preCheck
+
+    meson test --timeout-multiplier 4 --print-errorlogs
```

Comment:
Why not use https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/me/meson/setup-hook.sh#L70 ? 
Maybe with dontAddTimeoutMultiplier=true and mesonCheckFlags=(--timeout-multiplier 4)

Reply:
Thanks, removed the `checkPhase`, it predated `dontAddTimeoutMultiplier` and `mesonCheckFlags`.